packages/admin-ui/src/NoMatch.tsx

```diff
@@ -1,19 +1,18 @@
 import { Component } from 'react';
-import PropTypes from 'prop-types';
 
 import routes from './routes';
 
-const NoMatch = class NoMatch extends Component {
-  static propTypes = {
-    location: PropTypes.shape({
-      pathname: PropTypes.string.isRequired,
-    }).isRequired,
-  };
+export interface Props {
+  location: {
+    [pathname: string]: string,
+  },
+}
 
+class NoMatch extends Component<Props> {
   componentDidMount() {
     // We want to redirect everything without a match back to Drupal.
     if (!Object.keys(routes).includes(this.props.location.pathname)) {
-      window.location =
+      window.location.href =
```

Comment:
Isn't that doing the same?

Reply:
Apparently not https://stackoverflow.com/questions/13106950/set-window-location-with-typescript

Reply:
Oh I see, this is TS speciality

Reply:
Thank you